src/testUtils.ts

```diff
@@ -117,33 +126,42 @@ export function getBenchmarkFunctions(doc: vscode.TextDocument, token: vscode.Ca
  */
 export function goTest(testconfig: TestConfig): Thenable<boolean> {
 	return new Promise<boolean>((resolve, reject) => {
-		outputChannel.clear();
+
+		// We only want to clear the outputChannel if there are no tests in flight.
+		// We do not want to clear it if tests are already running, as that could
+		// lose valuable output.
+		if (testProcesses.length < 1) {
+			outputChannel.clear();
+		}
+
 		if (!testconfig.background) {
+			let buildTags: string = testconfig.goConfig['buildTags'];
+			let args: Array<string> = ['test', ...testconfig.flags];
+			let testType: string = testconfig.isBenchmark ? 'Benchmarks' : 'Tests';
 
+			if (testconfig.isBenchmark) {
+				args.push('-benchmem', '-run=^$');
+			} else {
+				args.push('-timeout', testconfig.goConfig['testTimeout']);
+			}
+			if (buildTags && testconfig.flags.indexOf('-tags') === -1) {
+				args.push('-tags', buildTags);
+			}
+
+			let testEnvVars = getTestEnvVars(testconfig.goConfig);
+			let goRuntimePath = getGoRuntimePath();
 			outputChannel.show(true);
 		}
 
 		let buildTags: string = testconfig.goConfig['buildTags'];
-		let args: Array<string> = ['test', ...testconfig.flags];
-		let testType: string = testconfig.isBenchmark ? 'Benchmarks' : 'Tests';
-
-		if (testconfig.isBenchmark) {
-			args.push('-benchmem', '-run=^$');
-		} else {
-			args.push('-timeout', testconfig.goConfig['testTimeout']);
-		}
+		let args = ['test', ...testconfig.flags, '-timeout', testconfig.goConfig['testTimeout']];
```

Comment:
This will apply the timeout when running benchmarks as well. I am guessing when benchmarks are run, they should be allowed to take as much time as they need and shouldnt be bound with the timeout.